cmds/exp/tsort/queue.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2012-2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+type queue struct {
```

Comment:
type queue []string may simplify this as well

Reply:
I'm sort of ok either way. If you use a struct, then, later, it is easy to add to it without changing existing code? wdyt?

Reply:
I agree, if there are plans to add something here, sure struct will work better.

Reply:
I was in two minds, as `queue`, `set` and `multiset` are internal data structures. But I imagine they could be used elsewhere in the future, so I'm inclined to leave them be. WDYT?

Reply:
Note: I purposefully made `set` extend `map[string]struct{}`, rather than encapsulate it in a struct, because AFAICT none of the operations that one could do on a map can invalidate a set. For example, all map keys are unique, therefore all set values are unique too. But I'm happy to be challenged on this.

Reply:
It's fine by me, I don't have any strong preference here. Also I guess if Go standard library will catch up with generics all of this will be gone.

Reply:
Perfect. In that case I'll set this conversation to "resolved". Thanks, both!